processing/data_collection/gazette/spiders/sc_blumenau.py

```diff
@@ -0,0 +1,56 @@
+
+from dateparser import parse
+from datetime import datetime
+from scrapy import Request
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class ScBlumenauSpider(BaseGazetteSpider):
+    TERRITORY_ID = '4202404'
+    NEXT_PAGE_URL = 'https://www.diariomunicipal.sc.gov.br{}'
+    POSSIBLE_GAZETTE_CSS = '.quiet'
+    NEXT_PAGE_CSS = '.pagination li.next:not(.disabled) a::attr(href)'
+
+    allowed_domains = ['diariomunicipal.sc.gov.br']
+    name = 'sc_blumenau'
+    start_urls = ['https://www.diariomunicipal.sc.gov.br/site/?r=site/index&q=cod_municipio%3A164']
+
+    def parse(self, response):
+        """
+        @url https://www.diariomunicipal.sc.gov.br/site/?r=site/index&q=cod_municipio%3A164
+        @returns requests 1
+        @scrapes date file_urls is_extra_edition territory_id power scraped_at
+        """
+
+        possible_gazettes = response.css(self.POSSIBLE_GAZETTE_CSS);
+        for element in possible_gazettes:
+            url = self.extract_url(element)
+            if url:
+                date = self.extract_date(element)
+
+                yield Gazette(
+                    date=date,
+                    file_urls=[url],
+                    is_extra_edition=False,
+                    territory_id=self.TERRITORY_ID,
+                    power='executive_legislature',
+                    scraped_at=datetime.utcnow(),
+                )
+        
+        next_page_url = self.extract_next_page_url(response)
```

Comment:
Removing the extra method will reduce the number of lines and make it easier to understand the code.
```
    next_page_path = response.css(self.NEXT_PAGE_CSS).extract_first()
    if next_page_path:
        yield response.urljoin(next_page_path)
```

When you receive a relative URL (for example "/site/?r=site/index&amp;q=cod_municipio%3A164&amp;Search_page=2"), you don't need to do string concatenation. Use the `response.urljoin` method available in Scrapy.

So replace `yield self.NEXT_PAGE_URL.format(next_page_path)` to `yield response.urljoin(next_page_path)`

Reply:
Didn't know urljoin. I will make the changes, thanks